generators/newfpmpage/index.js

```diff
@@ -0,0 +1,117 @@
+const path = require("path");
+const Generator = require("yeoman-generator");
+const fpmWriter = require("@sap-ux/fe-fpm-writer");
+const serviceWriter = require("@sap-ux/odata-service-writer");
+const axios = require("@sap-ux/axios-extension");
+
+module.exports = class extends Generator {
+    static displayName = "Enables the Fiori elements flexible program model";
+
+    async prompting() {
+        const modules = this.config.get("uimodules") || [];
+        
+        this.answers = await this.prompt([
+            {
+                type: "list",
+                name: "moduleName",
+                message: "To which module do you want add a custom FPM page?",
+                choices: modules,
+                when: !!modules && modules.length > 1
+            },
+            {
+                type: "input",
+                name: "viewName",
+                message: "What is the name of the page view?",
+                validate: (s) => {
+                    if (/^\d*[a-zA-Z][a-zA-Z0-9]*$/g.test(s)) {
+                        return true;
+                    }
+                    return "Please use alpha numeric characters only for the view name.";
+                },
+                default: 'Main'
+            },
+            
+        ]);
+
+        const manifest = this.fs.readJSON(this.destinationPath(this.answers.moduleName || this.options.modulename, 'webapp', 'manifest.json'));
+        if (!manifest) {
+            this.answers.serviceUrl = (await this.prompt([{
+                type: 'input',
+                name: 'serviceUrl',
+                message: 'What is the url of the main service?',
+                default: 'https://iccsrm.sap.com:44300/sap/opu/odata4/iwbep/v4_sample/default/iwbep/v4_gw_sample_basic/0001',
```

Comment:
Is it allowed to use this service officially? When googling I just found a hit to the server itself but no mentioning in any documentation.

Reply:
The service is used for openSAP courses but I will remove it before finishing this PR. I just let in there for quicker debugging